### PR TITLE
User role self upgrade

### PR DESCRIPTION
**Pull request by pruthvichivukula**

### CODE REVIEW

File: src/main/java/SpringJPA/PageController.java [hunk 1]
```diff
@@ -99,6 +109,41 @@ public String makeAPICall(Principal principal, @ModelAttribute String placeholde
...
+        User user = userRepository.findByUsername(principal.getName());
+        if (user.getRole() == UserType.ROLE_NONTRIAL && !user.getHasUsedTrial()){
+            user.setRole(UserType.ROLE_TRIAL);
+            user.startTrial();
+            SimpleGrantedAuthority authority = new SimpleGrantedAuthority(user.getRole().toString());
+            Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+            List<GrantedAuthority> updatedAuthorities = new ArrayList<>(auth.getAuthorities());
+            updatedAuthorities.add(authority);
+            Authentication newAuth = new UsernamePasswordAuthenticationToken(auth.getPrincipal(), auth.getCredentials(), updatedAuthorities);
+            SecurityContextHolder.getContext().setAuthentication(newAuth);
```
Comment (Justin-JKim) [inline src/main/java/SpringJPA/PageController.java]: I noticed some code repetition here, perhaps it should be placed into a function.

File: src/main/java/SpringJPA/PageController.java [hunk 2]
```diff
@@ -99,6 +109,41 @@ public String makeAPICall(Principal principal, @ModelAttribute String placeholde
...
+    public String makePaidUserRequest(Principal principal, @ModelAttribute String placeholder){
+        User user = userRepository.findByUsername(principal.getName());
+        if (user.getRole() == UserType.ROLE_NONTRIAL || user.getRole() == UserType.ROLE_TRIAL){
+            user.setRole(UserType.ROLE_PREMIUM);
+            SimpleGrantedAuthority authority = new SimpleGrantedAuthority(user.getRole().toString());
+            Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+            List<GrantedAuthority> updatedAuthorities = new ArrayList<>(auth.getAuthorities());
+            updatedAuthorities.add(authority);
+            Authentication newAuth = new UsernamePasswordAuthenticationToken(auth.getPrincipal(), auth.getCredentials(), updatedAuthorities);
+            SecurityContextHolder.getContext().setAuthentication(newAuth);
```
Comment (Justin-JKim) [inline src/main/java/SpringJPA/PageController.java]: I noticed some code repetition here, perhaps it should be placed into a function.

**Review by Justin-JKim — CHANGES_REQUESTED** [review]
You should place repetitive code blocks in PageController on line 119-124 and 135-140 into a function. Otherwise it looks good.

File: src/main/java/SpringJPA/PageController.java (hunk 1 above)
Comment (pruthvichivukula) [reply src/main/java/SpringJPA/PageController.java]: pushed an update

File: src/main/java/SpringJPA/PageController.java (hunk 2 above)
Comment (pruthvichivukula) [reply src/main/java/SpringJPA/PageController.java]: pushed an update

**Review by Justin-JKim — APPROVED** [review]
updateAuthWhenUpgrading function looks good.

**Pull request merged**